pandas/core/reshape/reshape.py

```diff
@@ -1094,6 +1097,208 @@ def get_empty_frame(data) -> DataFrame:
+        character indicating the separation of the categorical names from the prefixes.
+        For example, if your column names are 'prefix_A' and 'prefix_B',
+        you can strip the underscore by specifying sep='_'.
+        Alternatively, pass a dictionary to map prefix separators to prefixes if
+        multiple and/or mixed separators are used in the column names.
+    dropped_fist : None, Hashable or dict of Hashables, default None
+        The implied value the dummy takes when all values are zero.
+        Can be a a single value for all variables or a dict directly mapping the
+        dropped value to a prefix of a variable.
+
+    Returns
+    -------
+    `DataFrame`
+        Categorical data decoded from the dummy input-data.
+
+    See Also
+    --------
+    get_dummies : Convert `Series` or `DataFrame` to dummy codes.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0],
+    ...                   "c": [0, 0, 1, 0]})
+
+    >>> pd.from_dummies(df)
+    0     a
+    1     b
+    2     c
+    3     a
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 1], "col1_b": [0, 1, 0],
+    ...                   "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                   "col2_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep="_")
+        col1    col2
+    0    a       b
+    1    b       a
+    2    a       c
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 0], "col1_b": [0, 1, 0],
+    ...                    "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                    "col2_c": [0, 0, 0]})
+
+    >>> pd.from_dummies(df, sep="_", dropped_first={"col1": "d", "col2": "e"})
+        col1    col2
+    0    a       b
+    1    b       a
+    2    d       e
+
+    >>> df = pd.DataFrame({"col1_a-a": [1, 0, 1], "col1_b-b": [0, 1, 0],
+    ...                   "col2-a_a": [0, 1, 0], "col2-b_b": [1, 0, 0],
+    ...                   "col2-c_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep={"col1": "_", "col2": "-"})
+       col1  col2
+    0  a-a   b_b
+    1  b-b   a_a
+    2  a-a   c_c
+    """
+    from pandas.core.reshape.concat import concat
+
+    if subset is None:
+        subset = list(data.columns)
+    elif isinstance(subset, Index):
+        subset = list(subset)
+    elif not isinstance(subset, list):
+        raise TypeError(
+            f"Expected 'subset' to be of type 'Index', or 'list'; "
+            f"Received 'subset' of type: {type(subset).__name__}"
+        )
+
+    if data[subset].isna().any().any():
+        raise ValueError(
+            f"Dummy DataFrame contains NA value in column: "
+            f"'{data.columns[data.isna().any().argmax()]}'"
+        )
+
+    # index data with a list of all columns that are dummies
+    try:
+        data_to_decode = data[subset].astype("boolean")
+    except TypeError:
+        raise TypeError("Passed DataFrame contains non-dummy data")
+
+    # get separator for each prefix and lists to slice data for each prefix
```

Comment:
umm this is *very* complicated. what are you actually trying to do here?

Reply:
I want to get all columns that correspond to a specific prefix such that I can extract the values for each block. I do this here to avoid deep nesting (and checking whether or not a column belongs to a prefix) later on, when the value for each entry is determined.